docs/proposals/20191106-machine-load-balancer.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Machine Load Balancer 
+authors:
+  - "@moshloop"
+reviewers:
+    - "@akutz"
+    - "@ncdc"
+    - "@timothysc"
+    - "@vincepri"
+creation-date: 2019-11-06
+last-updated: 2019-11-06
+status: provisional
+---
+
+# Machine Load Balancer
+
+### Motivation
+
+* On-Premises clusters (without access to a Public Cloud Load Balancer as a Service) need a pluggable mechanism to enable the deployment of highly-available (HA) control planes using Cluster API (CAPI).
+
+* Multi-cluster environments need a mechanism to load balance ingress across multiple clusters in a safe and deterministic manner to support clusters as cattle / immutable clusters
+
+* Some environments need the ability to create different load balancing topologies e.g. 1 LB for control plane traffic, and another for external traffic to the kubernetes API. Or 1 LB per AZ to reduce inter-AZ traffic.
```

Comment:
This seems like an implementation detail or perhaps even out-of-scope? Shouldn't we focus on the simplest case/happy path for ingress API server access first?